pkg/metrics/metrics.go

```diff
@@ -36,31 +36,38 @@ var (
 	)
 )
 
-// SetCompletedState set completed states
-func SetCompletedState(specialResource string, state string, value int) {
-	completedStates.WithLabelValues(specialResource, state).Set(float64(value))
+func init() {
+	// Register custom metrics with the global prometheus registry
+	metrics.Registry.MustRegister(
+		createdSpecialResources,
+		completedStates,
+		completedKinds,
+	)
 }
 
-// DeleteCompleteStates delete metric complete states
-func DeleteCompleteStates(specialResource string, state string) {
-	completedStates.DeleteLabelValues(specialResource, state)
+//go:generate mockgen -source=metrics.go -package=metrics -destination=metrics_mock_test.go
```

Comment:
How about using the same name format as in @yevgeny-shnaidman's PR? https://github.com/openshift-psap/special-resource-operator/pull/155/files#diff-0f9d1b07bfdacdbad090371cd1eccaa2ebdfbb8d0b460fe12cec8443dbf0ae37R31

Reply:
I thought about it, but I think it should end with `_test.go`. Otherwise it'll be compiled into prod binary and counted toward coverage

Reply:
But then I don't think we can import it from another test package?

Reply:
That is correct. Thanks for catching that. Changed to be consistent with Yevgeny's code